WeakAuras/Profiling.lua

```diff
@@ -459,250 +404,107 @@ function WeakAuras.PrintProfile()
-        self:SetPoint("RIGHT", self.parent.barsFrame, "RIGHT")
-        if pos % 2 == 0 then
-          bar.fg:SetColorTexture(0.7, 0.7, 0.7, 0.7)
-          bar.bg:SetColorTexture(0, 0, 0, 0.2)
-        else
-          bar.fg:SetColorTexture(0.5, 0.5, 0.5, 0.7)
-          bar.bg:SetColorTexture(0, 0, 0, 0.4)
-        end
-        self:Show()
-      end
-    end
-
-    return bar
+    frame.progressBar:SetStatusBarColor(0.5, 0.5, 0.5, 0.7)
   end
 end
 
-function RealTimeProfilingWindow:RefreshBars()
-  if not profileData.systems.time or profileData.systems.time.count == 0 then
-    return
-  end
+local modes = {
+  L["Auras"],
+  L["Systems"]
+}
+WeakAurasProfilingMixin = {}
 
-  local total = TotalProfileTime(profileData.auras)
-  for i, name in ipairs(SortProfileMap(profileData.auras)) do
-    if (name ~= "time" and name ~= "wa") then
-      local bar = self:GetBar(name)
-      local elapsed = profileData.auras[name].elapsed
-      local pct = 100 * elapsed / total
-      local spike = profileData.auras[name].spike
-      bar:SetPosition(i)
-      bar:SetProgress(pct)
-      bar:SetText(elapsed, pct, spike)
-    end
-  end
-  if profileData.systems.wa then
-    local timespent = debugprofilestop() - profileData.systems.time.start
-    self.statsFrameText:SetText(("|cFFFFFFFFTime in WA: %.2fs / %ds (%.1f%%)"):format(
-      profileData.systems.wa.elapsed / 1000,
-      timespent / 1000,
-      100 * profileData.systems.wa.elapsed / timespent
-    ))
-  end
-end
+function WeakAurasProfilingMixin:OnLoad()
+  ButtonFrameTemplate_HidePortrait(self)
+  self:SetTitle(L["WeakAuras Profiling"])
+  self:SetSize(500, 300)
+  self.mode = 1
+  UIDropDownMenu_SetText(self.buttons.modeDropDown, modes[1])
 
-function RealTimeProfilingWindow:ResetBars()
-  for k, v in pairs(self.bars) do
-    v:Hide()
-  end
-end
-
-function RealTimeProfilingWindow:Init()
-  self:ClearAllPoints()
-  self:SetSize(self.width, self.height)
-  self:SetClampedToScreen(true)
-
-  if WeakAurasSaved.RealTimeProfilingWindow then
-    self:SetPoint("TOPLEFT", UIParent, "TOPLEFT",
-                  WeakAurasSaved.RealTimeProfilingWindow.xOffset or 0,
-                  WeakAurasSaved.RealTimeProfilingWindow.yOffset or 0)
-  else
-    self:SetPoint("TOPLEFT", UIParent, "TOPLEFT")
-  end
-  self:Show()
-
-  WeakAurasRealTimeProfilingTitleText:SetText(L["WeakAuras Profiling"])
-
-  local barsFrame = CreateFrame("Frame", nil, self)
-  self.barsFrame = barsFrame
-  barsFrame:SetPoint("TOPLEFT", 7, -20)
-  barsFrame:SetPoint("BOTTOMRIGHT", -3, 30)
-  barsFrame:Show()
-
-  local statsFrameText = self:CreateFontString(nil, "OVERLAY", "GameFontNormal")
-  self.statsFrameText = statsFrameText
-  statsFrameText:SetPoint("BOTTOMLEFT", 15, 25)
+  self.buttons.report:SetText(L["Report Summary"])
+  self.buttons.report.tooltip = L["A detailed overview of your auras and WeakAuras systems\nCopy the whole text to Weakaura's Discord if you need assistance."]
```

Comment:
I think this is a bit too hidden now.

Reply:
This is still shown in the text summary, it's not more hidden than it used to be